go/example.go

```diff
@@ -96,47 +89,53 @@ func main() {
 	client, err := flight.NewClientWithMiddleware(
 		net.JoinHostPort(config.Host, config.Port),
 		nil,
-		[]flight.ClientMiddleware{
-			flight.NewClientCookieMiddleware(),
-		},
+		[]flight.ClientMiddleware{flight.NewClientCookieMiddleware()},
 		grpc.WithTransportCredentials(creds),
 	)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer client.Close()
 
+	if err := run(config, client, interfaces.WrapRecordReader); err != nil {
```

Comment:
Will it just work with something like:

if err := run(config, client, flight.NewRecordReader); err != nil {

?

So then we can get rid of wrapReader as well?

Reply:
It will probably run but the problem isn't running, its the tests. flight.NewRecordReader doesn't satisfy any interface, that's why I created the WrapRecordReader so I could mock it. Haven't being able to find a workaround that works.

Reply:
I can also put WrapRecordReader in example.go that way its not in another file

Reply:
Yes, I mean can we
1) leave the RecordReader interface in interfaces.
2) Keep current run signature
```
func run(config interfaces.FlightConfig, client flight.Client,
	readerCreator func(flight.FlightService_DoGetClient) (interfaces.RecordReader, error),
) error {
```
3) In example.go pass inside flight.NewRecordReader (will go do here implicit conversion to the interfaces.RecordReader ?)
4) Use MockRecordReader in tests

Will that work?

And if doesn't work and we need to keep the WrapRecordReader, I think it is still better to move it it from interfaces here, since 1) it is used once only here and 2) I am not sure keeping implementation in a file with interface in the interface dir is stylish :)